Intermittent test execution failure for libandroid-navigation:testDebugUnitTest

CircleCI builds are intermittently failing with:

```
:libandroid-navigation:testDebugUnitTest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':libandroid-navigation:testDebugUnitTest'.
> Process 'Gradle Test Executor 1' finished with non-zero exit value 137
  This problem might be caused by incorrect test process configuration.
  Please refer to the test execution section in the user guide at https://docs.gradle.org/4.6/userguide/java_plugin.html#sec:test_execution
```

Per https://stackoverflow.com/questions/38967991/why-are-my-gradle-builds-dying-with-exit-code-137, this looks to be a memory issue that we can configure with `circle.yml` and our gradle files.

--- a/circle.yml
+++ b/circle.yml
@@ -21,6 +21,7 @@ jobs:
       JVM_OPTS: -Xmx3200m
       BUILDTYPE: Debug
       IS_LOCAL_DEVELOPMENT: false
+      GRADLE_OPTS: -Xmx2048m -Dorg.gradle.daemon=false
     steps:
       - checkout
       - restore_cache:
@@ -38,7 +39,7 @@ jobs:
       - run:
           name: Run unit-test, generate Jacoco test report and Post code coverage report to Codecov.io
           command: |
-            ./gradlew testDebugUnitTestCoverage --no-daemon
+            ./gradlew testDebugUnitTestCoverage
             pip install --user codecov && /root/.local/bin/codecov
       - run:
           name: Build release to test ProGuard rules

